scripts/shared.sh

```diff
@@ -4,11 +4,94 @@
 set -a
 set -eu -o pipefail
 
+function _extract_from_file() {
+    local key="$1"
+    local config_file="$2"
+    local value
+    if [[ -f $config_file ]]; then
+        value="$(eval "jq -r .${key} $config_file")"
+        if [[ "$value" != "null" ]]; then
+            echo "$value"
+            return 0
+        fi
+    fi
+}
+
 function _extract_cluster_config_value() {
     # Try extracting config from cluster config.json, default config.json, then aladdin config.json
+    local key="$1"
     local value
-    value="$1"
-    jq -nr --arg value "$value" 'first(inputs | (if .[$value] == null then empty else .[$value] end))' \
-        "$ALADDIN_CONFIG_DIR/$CLUSTER_CODE/config.json" "$ALADDIN_CONFIG_DIR/default/config.json" \
-        "$ALADDIN_CONFIG_DIR/config.json"
+    value=$(_extract_from_file "$key" "$ALADDIN_CONFIG_DIR/$CLUSTER_CODE/config.json")
+    if ! test -z "$value"; then
+        echo "$value"
+        return 0
+    fi
+    value=$(_extract_from_file "$key" "$ALADDIN_CONFIG_DIR/default/config.json")
+    if ! test -z "$value"; then
+        echo "$value"
+        return 0
+    fi
+    value=$(_extract_from_file "$key" "$ALADDIN_CONFIG_DIR/config.json")
+    if ! test -z "$value"; then
+        echo "$value"
+        return 0
+    fi
+}
```

Comment:
improved this function to be able to get nested values across several config files

Reply:
Could it not before?

Reply:
before it could only get flat values merged across configs iirc, now it can also get nested values

Reply:
does anything use this nested lookup yet?

Reply:
not yet